Generator/AssociationGenerator.php

```diff
@@ -0,0 +1,269 @@
+            }
+
+            $this->bufferizeAssociation($association, $tmpFile);
+
+            $this->objectDetacher->detach($product);
+
+            $progress->advance();
+        }
+
+        $this->writeCsvFile($this->headers, $outputFile, $tmpFile, $delimiter);
+        unlink($tmpFile);
+
+        return $this;
+    }
+
+    /**
+     * Generates a list of random comma separated association types code
+     *
+     * @param $typeOfAssociation string
+     *
+     * @return array
+     */
+    protected function getRandomAssociations($typeOfAssociation)
+    {
+        $associationTypes = $this->faker->randomElements($this->associationTypeRepository->findAll());
+        $associationTypesCodes = array_map(function ($associationType) use ($typeOfAssociation) {
+            return $associationType->getCode() . $typeOfAssociation;
+        }, $associationTypes);
+
+        return join(',', $associationTypesCodes);
+    }
+
+    /**
+     * Find product skus different from the product passed as parameter
+     *
+     * @param ProductInterface $excludedProduct
+     * @param CursorInterface  $products
+     * @param int              $associationCount
+     *
+     * @return array
+     */
+    protected function generateRandomProducts($excludedProduct, CursorInterface $products, $associationCount)
+    {
+        $productIdentifiers = [];
+        $i = 0;
+
+        while ($i < $associationCount) {
+            if (!$products->valid()) {
+                $products->rewind();
+                $products->next();
+            }
+
+            $product = $products->current();
+
+            if ($excludedProduct->getIdentifier()->getData() !== $product->getIdentifier()->getData()) {
+                $productIdentifiers[] = $product->getIdentifier()->getData();
+                $i++;
+            }
+
+            $products->next();
+
+            $this->objectDetacher->detach($product);
+        }
+
+        return join(',', $productIdentifiers);
+    }
+
+    /**
+     * Generate a comma separated list of random variant groups codes
+     *
+     * @param $associationsVariantGroupCount
+     *
+     * @return string
+     */
+    private function generateRandomVariantGroups($associationsVariantGroupCount)
+    {
+        if (0 === $associationsVariantGroupCount) {
+            return '';
+        }
+
+        $variantGroups = $this->faker->randomElements(
+            $this->groupRepository->getAllVariantGroups(),
+            $associationsVariantGroupCount
+        );
+
+        $variantGroupCodes = array_map(function ($variantGroup) {
+            return $variantGroup->getCode();
+        }, $variantGroups);
+
+        return join(',', $variantGroupCodes);
+    }
+
+    /**
+     * @return CursorInterface
+     */
+    private function getProducts()
+    {
+        $productQueryBuilder = $this->productQueryBuilderFactory->create();
+        $products = $productQueryBuilder->execute();
+
+        return $products;
```

Comment:
Why would it ? We use the productQueryBuilder that batches the queries (by sets of 100).
Thats what we used everywhere in the PIM to iterate over products.

Do you have any case in my mind the pqb didn't work ?